src/Microsoft.DotNet.ProjectModel/OutputPathCalculator.cs

```diff
@@ -61,5 +74,58 @@ public string GetIntermediateOutputPath(string buildConfiguration, string interm
 
             return intermediateOutputPath;
         }
+        
+        public string GetAssemblyPath(string buildConfiguration)
+        {
+            var compilationOptions = _project.GetCompilerOptions(_framework, buildConfiguration);
+            var outputExtension = FileNameSuffixes.DotNet.DynamicLib;
+
+            if (_framework.IsDesktop() && compilationOptions.EmitEntryPoint.GetValueOrDefault())
+            {
+                outputExtension = FileNameSuffixes.DotNet.Exe;
+            }
+
+            return Path.Combine(
+                GetOutputDirectoryPath(buildConfiguration),
+                _project.Name + outputExtension);
+        }
+        
+        public IEnumerable<string> GetBuildOutputs(string buildConfiguration)
```

Comment:
Very useful for incremental IO computation. Can you please edit GetCompilerIO to use GetBuildOutputs instead of having this logic duplicated? Or I could do it in my bugfix PR, after your PR goes in.